presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergConnector.java

```diff
@@ -180,9 +179,11 @@ public ConnectorAccessControl getAccessControl()
     public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel, boolean readOnly)
     {
         checkConnectorSupports(READ_COMMITTED, isolationLevel);
-        ConnectorTransactionHandle transaction = new HiveTransactionHandle();
+        IcebergTransactionHandle transaction = new IcebergTransactionHandle();
         try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
-            transactionManager.put(transaction, metadataFactory.create());
+            IcebergMetadata icebergMetadata = metadataFactory.create();
+            icebergMetadata.setTransactionHandle(transaction);
```

Comment:
Pass transaction to the `create()` call so that the field can be final

Reply:
Actually, if we move the state into `IcebergMetadata`, we shouldn't need the handle at all there (see other comment).

Reply:
Yes, done